### PR TITLE
MS Teams - Adding ring-user command - reopened again

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22329

## Description
continuing https://github.com/demisto/content/pull/6143  and https://github.com/demisto/content/pull/6164 after corruption...
Added ring user command and added micrisoft-teams suffix to commands

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 1]
```diff
@@ -1495,6 +1495,78 @@ def messages() -> Response:
...
+    """
+    bot_id = demisto.params().get('bot_id')
+    integration_context: dict = demisto.getIntegrationContext()
+    tenant_id: str = integration_context.get('tenant_id', '')
+    if not tenant_id:
+        raise ValueError(
+            'Did not receive tenant ID from Microsoft Teams, verify the messaging endpoint is configured correctly.'
+        )
+
+    callback_uri = demisto.args().get('callback_url', 'https://callback.url')
```
Comment (ronykoz) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]: didn't we say we are removing this?

File: Integrations/MicrosoftTeams/MicrosoftTeams.yml [hunk 2]
```diff
@@ -221,11 +221,77 @@ script:
...
+      name: username
+      required: true
+      secret: false
+    - default: false
+      description: The callback URL for the call for the bot who will recive a callback
+        from the generated ring. This is the same URL used when registering the calling bot (more info here https://docs.microsoft.com/en-us/microsoftteams/platform/bots/calls-and-meetings/registering-calling-bot#add-microsoft-graph-permissions)
+      isArray: false
+      name: callback_url
+      required: false
+      secret: false
```
Comment (ronykoz) [inline Integrations/MicrosoftTeams/MicrosoftTeams.yml]: remove this

Comment (roysagi) [reply Integrations/MicrosoftTeams/MicrosoftTeams.yml]: done

File: Integrations/MicrosoftTeams/MicrosoftTeams.py (hunk 1 above)
Comment (roysagi) [reply Integrations/MicrosoftTeams/MicrosoftTeams.py]: done

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 3]
```diff
@@ -1495,6 +1495,75 @@ def messages() -> Response:
...
+        )
+    # get user to call name and id
+    username_to_call = demisto.args().get('username')
+    users: list = get_users()
+    user_id: str = str()
+    for user in users:
+        if username_to_call in {user.get('displayName', ''), user.get('mail'), user.get('userPrincipalName')}:
+            user_id = user.get('id', '')
+            break
+    if len(user_id) == 0:
```
Comment (Itay4) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]:
```suggestion
    if not used_id:
```

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 4]
```diff
@@ -1495,6 +1495,75 @@ def messages() -> Response:
...
+        ],
+        "requestedModalities": [
+            "audio"
+        ],
+        "mediaConfig": {
+            "@odata.type": "#microsoft.graph.serviceHostedMediaConfig",
+        },
+        "tenantId": tenant_id
+    }
+    response = http_request(method='POST',
```
Comment (Itay4) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]: we should split this func to two: command and request
you are not supposed to call `http_request` from a command func

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 5]
```diff
@@ -1495,6 +1495,75 @@ def messages() -> Response:
...
+                            url=f'{GRAPH_BASE_URL}/v1.0/communications/calls',
+                            json_=call_request_data)
+
+    return_outputs("Calling {}".format(username_to_call), {}, response)
```
Comment (Itay4) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]: use f-strings instead of format

File: Integrations/MicrosoftTeams/MicrosoftTeams.yml [hunk 6]
```diff
@@ -221,11 +221,71 @@ script:
...
       secret: false
-    deprecated: false
+    deprecated: true
     description: Adds a member (user) to a private channel.
     execution: false
     name: add-user-to-channel
-  dockerimage: demisto/teams:1.0.0.1701
+  - arguments:
+    - default: false
+      description: The user to call.
```
Comment (Itay4) [inline Integrations/MicrosoftTeams/MicrosoftTeams.yml]: if i understand correctly, this is the display name? will email address work? need to clarify this in the description

**Pull request merged**